src/main/host/syscall/format.rs

```diff
@@ -193,16 +193,28 @@ impl SyscallPtrDisplay for SyscallPtr<*const i8> {
         // much at once (for example bytes past the null byte), we may try to read from an invalid
         // address
 
-        // get bytes until either null byte or a read error
+        // to avoid printing too many escaped bytes, limit the number of non-graphic and non-ascii
+        // characters
+        let mut non_graphic_remaining = DISPLAY_LEN / 3;
+
         let mut s: Vec<NonZeroU8> = reader
             .bytes()
+            // get bytes until either null byte or a read error
             .map_while(|x| x.ok().and_then(|y| NonZeroU8::new(y)))
+            // stop after a certain number of non-graphic characters
+            .map_while(|x| {
+                if !x.get().is_ascii_graphic() {
+                    non_graphic_remaining = non_graphic_remaining.saturating_sub(1);
+                }
+                (non_graphic_remaining > 0).then(|| x)
```

Comment:
[TIL](https://doc.rust-lang.org/std/primitive.bool.html#method.then)

Reply:
Yeah I don't know if this is too weird, but I think it's readable. (And it looks like `bool::then_some()` will be stabilized soon.)